appdaemon/plugins/hass/hassplugin.py

```diff
@@ -42,76 +42,41 @@ def __init__(self, ad: AppDaemon, name, args):
         self.config = args
         self.name = name
 
-        self.stopping = False
-        self.ws = None
-        self.reading_messages = False
-        self.metadata = None
-        self.hass_booting = False
-
         self.logger.info("HASS Plugin Initializing")
 
-        self.name = name
-
-        if "namespace" in args:
-            self.namespace = args["namespace"]
-        else:
-            self.namespace = "default"
-
+        # validate basic config
         if "ha_key" in args:
-            self.ha_key = args["ha_key"]
             self.logger.warning("ha_key is deprecated please use HASS Long Lived Tokens instead")
-        else:
-            self.ha_key = None
-
-        if "token" in args:
-            self.token = args["token"]
-        else:
-            self.token = None
-
-        if "ha_url" in args:
-            self.ha_url = args["ha_url"]
-        else:
+        if "ha_url" not in args:
             self.logger.warning("ha_url not found in HASS configuration - module not initialized")
 
-        if "cert_path" in args:
-            self.cert_path = args["cert_path"]
-        else:
-            self.cert_path = None
-
-        if "timeout" in args:
-            self.timeout = args["timeout"]
-        else:
-            self.timeout = None
-
-        if "retry_secs" in args:
-            self.retry_secs = int(args["retry_secs"])
-        else:
-            self.retry_secs = 5
-
-        if "cert_verify" in args:
-            self.cert_verify = args["cert_verify"]
-        else:
-            self.cert_verify = True
-
-        if "commtype" in args:
-            self.commtype = args["commtype"]
-        else:
-            self.commtype = "WS"
-
-        if "appdaemon_startup_conditions" in args:
-            self.appdaemon_startup_conditions = args["appdaemon_startup_conditions"]
-        else:
-            self.appdaemon_startup_conditions = {}
-
-        if "plugin_startup_conditions" in args:
-            self.plugin_startup_conditions = args["plugin_startup_conditions"]
-        else:
-            self.plugin_startup_conditions = {}
+        # Locally store common args and their defaults
+        self.appdaemon_startup_conditions = args.get("appdaemon_startup_conditions", {})
+        self.cert_path = args.get("cert_path")
+        self.cert_verify = args.get("cert_verify")
+        self.commtype = args.get("commtype", "WS")
+        self.ha_key = args.get("ha_key")
+        self.ha_url = args.get("ha_url", "")
```

Comment:
This is a behavior change.  Previously, if you didn't specify the ha_url, it would crash on startup with "no attribute 'ha_url'".  Now it will fail to connect instead.